odxtools/parameters/nrcconstparameter.py

```diff
@@ -88,49 +88,40 @@ def is_settable(self) -> bool:
-            if physical_value not in self.coded_values:
-                odxraise(
-                    f"The value of parameter '{self.short_name}' must "
-                    f" be one of {self.coded_values} (is: {physical_value!r})", EncodeError)
-                coded_value = self.coded_values[0]
-            else:
-                coded_value = physical_value
-        else:
-            # If the user did not select a value, the value of the
-            # this parameter is set by another parameter which
-            # overlaps with it. We thus just move the cursor.
-            bit_pos = encode_state.cursor_bit_position
-            bit_len = self.diag_coded_type.get_static_bit_length()
-
-            if bit_len is None:
-                odxraise("The diag coded type of NRC-CONST parameters must "
-                         "exhibit a static size")
-                return
-
-            encode_state.cursor_byte_position += (bit_pos + bit_len + 7) // 8
-            encode_state.cursor_bit_position = 0
+            odxraise("The value of NRC-CONST parameters cannot be set directly!", EncodeError)
+
+        # TODO (?): extract the parameter and check if it is one of
+        # the values of self.coded_values. if not, throw an
+        # EncodeMismatch exception! This is probably a bad idea
+        # because the parameter which determines the value of the
+        # NRC-CONST might possibly be specified after the NRC-CONST.
+
+        # move the cursor forward by the size of the parameter
+        bit_pos = encode_state.cursor_bit_position
+        bit_len = self.diag_coded_type.get_static_bit_length()
+
+        if bit_len is None:
+            odxraise("The diag coded type of NRC-CONST parameters must "
+                     "exhibit a static size")
             return
 
-        self.diag_coded_type.encode_into_pdu(cast(AtomicOdxType, coded_value), encode_state)
+        encode_state.cursor_byte_position += (bit_pos + bit_len + 7) // 8
+        encode_state.cursor_bit_position = 0
+
+        encode_state.emplace_bytes(b'', self.short_name)
 
     @override
     def _decode_positioned_from_pdu(self, decode_state: DecodeState) -> AtomicOdxType:
-        # Extract coded values
+        # Extract coded value
         coded_value = self.diag_coded_type.decode_from_pdu(decode_state)
 
         # Check if the coded value in the message is correct.
         if coded_value not in self.coded_values:
-            warnings.warn(
-                f"Coded constant parameter does not match! "
-                f"The parameter {self.short_name} expected a coded "
-                f"value in {str(self.coded_values)} but got {str(coded_value)} "
-                f"at byte position {decode_state.cursor_byte_position} "
-                f"in coded message {decode_state.coded_message.hex()}.",
-                DecodeError,
-                stacklevel=1,
-            )
+            raise DecodeMismatch(f"NRC-CONST parameter {self.short_name} does not apply")
```

Comment:
this is not really possible: we want to abort the decoding process for the whole response even regardless of whether we are in strict mode or not. This is because if this happens, the response is not applicable. Since this is IMO normal processing and not really exceptional circumstances, I'm not really happy about using exceptions for flow control. That said, any other solution I can think of is at least one order of magnitude more invasive and error-prone...

Reply:
(I have to correct myself: actually, going the warnings route and making `warnings` raise would be possible, but it would be very convoluted and IMO it also would defeat the purpose. In particular, because these conditions can happen during normal operation, i.e. it is not a circumstance which one needs/wants be warned about.)